api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirServiceRequestServiceImpl.java

```diff
@@ -40,9 +44,16 @@
 	private SearchQuery<TestOrder, ServiceRequest, FhirServiceRequestDao<TestOrder>, ServiceRequestTranslator<TestOrder>> searchQuery;
 	
 	@Override
-	public IBundleProvider searchForServiceRequests() {
-		SearchParameterMap theParams = new SearchParameterMap();
+	@Transactional(readOnly = true)
+	public IBundleProvider searchForServiceRequests(ReferenceAndListParam patientReference, TokenAndListParam code,
+	        ReferenceAndListParam encounterReference, ReferenceAndListParam participantReference, DateRangeParam occurence) {
 		
+		SearchParameterMap theParams = new SearchParameterMap()
+		        .addParameter(FhirConstants.ENCOUNTER_REFERENCE_SEARCH_HANDLER, encounterReference)
+		        .addParameter(FhirConstants.PATIENT_REFERENCE_SEARCH_HANDLER, patientReference)
+		        .addParameter(FhirConstants.CODED_SEARCH_HANDLER, code)
+		        .addParameter(FhirConstants.PARTICIPANT_REFERENCE_SEARCH_HANDLER, participantReference)
+		        .addParameter(FhirConstants.DATE_RANGE_SEARCH_HANDLER, "scheduledDate,dateActivated,dateStopped,autoExpireDate", occurence);
```

Comment:
Here, you can do something like we handle Person addresses. That brings more clarity to the code.

Reply:
@varung-31 Not sure I get your suggestion here since the parameter name that is used is only **occurence** unlike in personAddress where we have different name parameters..

Reply:
Okay, got your point. In this case, since these parameters are used only for ServiceRequest, we can avoid mentioning the properties in the SearchParameterMap for `FhirConstants.DATE_RANGE_SEARCH_HANDLER` and instead just use them directly in the handleDate() method that you have created. We won't have to add splitting of the property in different strings in that case.